### PR TITLE
Adds httpChecksumRequired trait

### PR DESCRIPTION
**Pull request by skotambkar**
*Description of changes:*
 - Adds support for httpChecksumRequired trait. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source/1.0/spec/core/behavior-traits.rst [hunk 1]
```diff
@@ -609,3 +609,30 @@ to request additional results from the operation.
...
+
+
+.. _httpChecksumRequired-trait:
+
+--------------------------------------
+``httpChecksumRequired`` trait
+--------------------------------------
+
+Summary
+    Indicates that an operation requires the checksum present in its HTTP
```
Comment (mtdowling) [inline docs/source/1.0/spec/core/behavior-traits.rst:621]: suggestion: "Indicates that an operation requires a checksum in its HTTP request. By default, the checksum used for a service is an MD5 passed in the `Content-MD5` header."

File: docs/source/1.0/spec/core/behavior-traits.rst [hunk 2]
```diff
@@ -609,3 +609,30 @@ to request additional results from the operation.
    presence, absence, or contents of a continuation token.
 
 .. _UUID: https://tools.ietf.org/html/rfc4122
+
+
+.. _httpChecksumRequired-trait:
+
+--------------------------------------
```
Comment (mtdowling) [inline docs/source/1.0/spec/core/behavior-traits.rst:616]: These headers are too long for the text

File: docs/source/1.0/spec/core/behavior-traits.rst [hunk 3]
```diff
@@ -609,3 +609,30 @@ to request additional results from the operation.
...
+``httpChecksumRequired`` trait
+--------------------------------------
```
Comment (mtdowling) [inline docs/source/1.0/spec/core/behavior-traits.rst:618]: The "-"'s need to line up with the text

File: docs/source/1.0/spec/core/behavior-traits.rst (hunk 1 above)
Comment (skotambkar) [reply docs/source/1.0/spec/core/behavior-traits.rst]: Updated with feedback

**Comment by skotambkar** [comment]
Updated with suggested feedback

**Pull request merged**